shpc/main/modules/base.py

```diff
@@ -331,152 +336,94 @@ def check(self, module_name):
-            if not os.path.exists(container_dest):
-                shutil.copyfile(container_path, container_dest)
-            container_path = container_dest
+        utils.mkdirp([module.module_dir, module.container_dir])
 
         # Add a .version file to indicate the level of versioning
-        self.versionfile.write(os.path.join(self.settings.module_base, uri), tag.name)
-
-        # For Singularity this is a path, podman is a uri. If None is returned
-        # there was an error and we cleanup
-        if not container_path:
-            container_path = self.container.registry_pull(
-                module_dir, container_dir, config, tag
-            )
-        if not container_path:
-            utils.remove_to_base(container_dir, self.container_base)
-            logger.exit("There was an issue pulling %s" % container_path)
+        self.versionfile.write(
+            os.path.join(self.settings.module_base, module.uri), module.tag.name
+        )
+        if not module.container_path:
+            utils.remove_to_base(module.container_dir, self.container_base)
+            logger.exit("There was an issue pulling the container for %s" % module.name)
 
         # Get the template based on the module and container type
         template = self.template.load(self.templatefile)
-        module_path = os.path.join(module_dir, self.modulefile)
+        module_path = os.path.join(module.module_dir, self.modulefile)
 
         # Install the container
-        self.container.install(
-            module_path,
-            container_path,
-            name,
-            template,
-            aliases=config.get_aliases(),
-            config=config,
-            parsed_name=config.name,
-            url=config.url,
-            description=config.description,
-            version=tag.name,
-            config_features=config.features,
-            features=kwargs.get("features"),
-        )
+        # This could be simplified to take the module
+        self.container.install(module_path, template, module, kwargs.get("features"))
 
         # If the container tech does not need storage, clean up
-        if not os.listdir(container_dir):
-            utils.remove_to_base(container_dir, self.container_base)
+        if not os.listdir(module.container_dir):
+            utils.remove_to_base(module.container_dir, self.container_base)
 
         # Write the environment file to be bound to the container
-        self.container.add_environment(
-            module_dir,
-            envars=config.get_envars(),
-            environment_file=self.settings.environment_file,
-        )
+        module.add_environment()
+        logger.info("Module %s was created." % module.tagged_name)
+        return module.container_path
 
-        if ":" not in name:
-            name = "%s:%s" % (name, tag.name)
-        logger.info("Module %s was created." % name)
+    def view_install(self, view_name, name, tag=None, force=False):
```

Comment:
This `force` controls 1 thing: whether we allow overwriting existing symlinks

Reply:
I really appreciate these comments about force (since the meaning varies between) could we add them to the function docstrings for us of  the future to see?